# Project:
**Exploratory Data Analysis (EDA) on Amazon Electronics Sales Dataset**

# Objective:

**Analyze Electronics sales dataset to derive insight and trends**

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))
        

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/amazon-products-dataset/Gaming Consoles.csv
/kaggle/input/amazon-products-dataset/Car Electronics.csv
/kaggle/input/amazon-products-dataset/Janitorial and Sanitation Supplies.csv
/kaggle/input/amazon-products-dataset/All Electronics.csv
/kaggle/input/amazon-products-dataset/All Books.csv
/kaggle/input/amazon-products-dataset/Make-up.csv
/kaggle/input/amazon-products-dataset/Travel Accessories.csv
/kaggle/input/amazon-products-dataset/Indian Language Books.csv
/kaggle/input/amazon-products-dataset/Car and Bike Care.csv
/kaggle/input/amazon-products-dataset/Sunglasses.csv
/kaggle/input/amazon-products-dataset/Bags and Luggage.csv
/kaggle/input/amazon-products-dataset/Yoga.csv
/kaggle/input/amazon-products-dataset/Sportswear.csv
/kaggle/input/amazon-products-dataset/Fiction Books.csv
/kaggle/input/amazon-products-dataset/Exam Central.csv
/kaggle/input/amazon-products-dataset/Home Storage.csv
/kaggle/input/amazon-products-dataset/Toys Gifting Store.csv
/kaggle/input/amazon-pr

In [2]:
# Load required libraries
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import plotly.express as px

In [3]:
# load DataFrame 
filepath='/kaggle/input/amazon-products-dataset/All Electronics.csv'
DS=pd.read_csv(filepath,low_memory=False)
DS.head(4)

name        main_category  \
0  Redmi 10 Power (Power Black, 8GB RAM, 128GB St...  tv, audio & cameras   
1  OnePlus Nord CE 2 Lite 5G (Blue Tide, 6GB RAM,...  tv, audio & cameras   
2  OnePlus Bullets Z2 Bluetooth Wireless in Ear E...  tv, audio & cameras   
3  Samsung Galaxy M33 5G (Mystique Green, 6GB, 12...  tv, audio & cameras   

      sub_category                                              image  \
0  All Electronics  https://m.media-amazon.com/images/I/81eM15lVcJ...   
1  All Electronics  https://m.media-amazon.com/images/I/71AvQd3Vzq...   
2  All Electronics  https://m.media-amazon.com/images/I/51UhwaQXCp...   
3  All Electronics  https://m.media-amazon.com/images/I/81I3w4J6yj...   

                                                link ratings no_of_ratings  \
0  https://www.amazon.in/Redmi-Power-Black-128GB-...     4.0           965   
1  https://www.amazon.in/OnePlus-Nord-Lite-128GB-...     4.3       113,956   
2  https://www.amazon.in/Oneplus-Bluetooth-Wirele...     4.2        90,304   
3  https://www.amazon.in/Samsung-Mystique-Storage...     4.1        24,863   

  discount_price actual_price  
0        ₹10,999      ₹18,999  
1        ₹18,999      ₹19,999  
2         ₹1,999       ₹2,299  
3        ₹15,999      ₹24,999

In [4]:
# check Dataframe size
DS.shape

(9600, 9)

In [5]:
# check missing values in Dataframe
DS.isna().sum()

name                0
main_category       0
sub_category        0
image               0
link                0
ratings            95
no_of_ratings      95
discount_price    484
actual_price       70
dtype: int64

In [6]:
# check unique values in Dataframe
DS.nunique()

name              8800
main_category        1
sub_category         1
image             9321
link              9600
ratings             39
no_of_ratings     3455
discount_price    1608
actual_price      1068
dtype: int64

In [7]:
# check columns in Dataframe
DS.columns

Index(['name', 'main_category', 'sub_category', 'image', 'link', 'ratings',
       'no_of_ratings', 'discount_price', 'actual_price'],
      dtype='object')

In [8]:
# check datatype of the Dataframe
DS.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9600 entries, 0 to 9599
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   name            9600 non-null   object
 1   main_category   9600 non-null   object
 2   sub_category    9600 non-null   object
 3   image           9600 non-null   object
 4   link            9600 non-null   object
 5   ratings         9505 non-null   object
 6   no_of_ratings   9505 non-null   object
 7   discount_price  9116 non-null   object
 8   actual_price    9530 non-null   object
dtypes: object(9)
memory usage: 675.1+ KB


# **Data Cleaning:**
**Here the datatypes of the columns *'name',  'main_category', 'sub_category', 'image', 'link', 'ratings', 'no_of_ratings', 'discount_price', 'actual_price'* are in object which is wrong.
So we have to change it to int or float.
Also the columns main_category and sub_category has multiple values so we will seperate them into indivisual columns**

In [9]:
#remove '₹' from columns 'actual_price' and 'discount_price'
DS["discount_price"] = DS["discount_price"].str.replace("₹","")
DS["actual_price"] = DS["actual_price"].str.replace("₹","")

In [10]:
#change datatype as mentioned
DS['discount_price'] = DS["discount_price"].str.replace(',', '').astype(float)
DS["actual_price"] = DS["actual_price"].str.replace(',', '').astype(float)

In [11]:
#change rating to normal 0.0 form
DS['ratings'].unique()

array(['4.0', '4.3', '4.2', '4.1', '3.9', '4.6', '3.8', 'Get', '4.4',
       '3.5', '3.7', '4.5', '3.3', '3.6', '3.4', nan, '3.2', '2.9', '4.7',
       '5.0', '2.5', '2.8', '4.8', '3.1', '3.0', '2.3', '2.7', '2.4',
       '4.9', '1.7', '1.5', '2.6', '1.8', '1.3', '2.1', '1.9', '2.2',
       '1.0', '1.4', '2.0'], dtype=object)

In [12]:
#extract digit and change it to float dtype
DS['ratings']=DS['ratings'].replace(['Get'], 0.0)
DS['ratings']=DS['ratings'].astype(float)

In [13]:
DS['ratings'].unique()

array([4. , 4.3, 4.2, 4.1, 3.9, 4.6, 3.8, 0. , 4.4, 3.5, 3.7, 4.5, 3.3,
       3.6, 3.4, nan, 3.2, 2.9, 4.7, 5. , 2.5, 2.8, 4.8, 3.1, 3. , 2.3,
       2.7, 2.4, 4.9, 1.7, 1.5, 2.6, 1.8, 1.3, 2.1, 1.9, 2.2, 1. , 1.4,
       2. ])

The 'no_of_ratings' column is converted to the float type in two steps: 
    first, a new boolean column is formed, where the True value corresponds to the numeric value
    in the original column.
    Then the values 'in the 'no_of_ratings' column are recalculated for values
    matching the True of the 'correct_no_of_ratings' column.

In [14]:
# Add column 'correct_no_of_ratings' which value is 'True' if 'no_of_ratings' begins from digit 
DS['no_of_ratings'] = DS['no_of_ratings'].astype(str)
DS['correct_no_of_ratings'] = pd.Series([DS['no_of_ratings'][x][0].isdigit() for x in range(len(DS['no_of_ratings']))])

# Drop columns with incorrect 'no_of_ratings'
DS = DS[DS['correct_no_of_ratings'] == True]
DS['correct_no_of_ratings'].value_counts()

correct_no_of_ratings
True    9481
Name: count, dtype: int64

In [15]:
# Change the type to float
DS["no_of_ratings"] = DS["no_of_ratings"].str.replace(',', '').astype(float)

In [16]:
DS.head()

name        main_category  \
0  Redmi 10 Power (Power Black, 8GB RAM, 128GB St...  tv, audio & cameras   
1  OnePlus Nord CE 2 Lite 5G (Blue Tide, 6GB RAM,...  tv, audio & cameras   
2  OnePlus Bullets Z2 Bluetooth Wireless in Ear E...  tv, audio & cameras   
3  Samsung Galaxy M33 5G (Mystique Green, 6GB, 12...  tv, audio & cameras   
4  OnePlus Nord CE 2 Lite 5G (Black Dusk, 6GB RAM...  tv, audio & cameras   

      sub_category                                              image  \
0  All Electronics  https://m.media-amazon.com/images/I/81eM15lVcJ...   
1  All Electronics  https://m.media-amazon.com/images/I/71AvQd3Vzq...   
2  All Electronics  https://m.media-amazon.com/images/I/51UhwaQXCp...   
3  All Electronics  https://m.media-amazon.com/images/I/81I3w4J6yj...   
4  All Electronics  https://m.media-amazon.com/images/I/71V--WZVUI...   

                                                link  ratings  no_of_ratings  \
0  https://www.amazon.in/Redmi-Power-Black-128GB-...      4.0          965.0   
1  https://www.amazon.in/OnePlus-Nord-Lite-128GB-...      4.3       113956.0   
2  https://www.amazon.in/Oneplus-Bluetooth-Wirele...      4.2        90304.0   
3  https://www.amazon.in/Samsung-Mystique-Storage...      4.1        24863.0   
4  https://www.amazon.in/OnePlus-Nord-Black-128GB...      4.3       113956.0   

   discount_price  actual_price  correct_no_of_ratings  
0         10999.0       18999.0                   True  
1         18999.0       19999.0                   True  
2          1999.0        2299.0                   True  
3         15999.0       24999.0                   True  
4         18999.0       19999.0                   True

In [17]:
DS.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9481 entries, 0 to 9599
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   name                   9481 non-null   object 
 1   main_category          9481 non-null   object 
 2   sub_category           9481 non-null   object 
 3   image                  9481 non-null   object 
 4   link                   9481 non-null   object 
 5   ratings                9481 non-null   float64
 6   no_of_ratings          9481 non-null   float64
 7   discount_price         9002 non-null   float64
 8   actual_price           9411 non-null   float64
 9   correct_no_of_ratings  9481 non-null   bool   
dtypes: bool(1), float64(4), object(5)
memory usage: 750.0+ KB


In [18]:
DS['ratings'].describe()

count    9481.000000
mean        4.077576
std         0.377189
min         1.000000
25%         3.900000
50%         4.100000
75%         4.300000
max         5.000000
Name: ratings, dtype: float64

Here, We can observe that the standard deviation is 0.37 and the mean is 4.07, indicating that the overall ratings for clothing products are above average. 
However, there are two possible scenarios to consider. Since 98.77 % of the products have been rated, it is likely that most of the customers have rated the product and those who were dissatisfied chose to leave a rating. 
Therefore, based on this analysis, it appears that **the overall condition of the Electronics Products is very much satisfactory from a customer perspective**.

In [19]:
# removing missing values from 'actual_price' & 'discount_price' column
DS=DS.dropna(subset=['actual_price','discount_price'])

In [20]:
#Extract the manufractor name from the dataset columns name
DS['manufacturer']=DS['name'].str.split(' ').str[0]
cols=DS.columns.tolist()
# change the position of manufracturer
cols=['name',
 'manufacturer',
 'main_category',
 'sub_category',
 'image',
 'link',
 'ratings',
 'no_of_ratings',
 'discount_price',
 'actual_price',
 'correct_no_of_ratings']
DS=DS[cols]
DS.head()

name manufacturer  \
0  Redmi 10 Power (Power Black, 8GB RAM, 128GB St...        Redmi   
1  OnePlus Nord CE 2 Lite 5G (Blue Tide, 6GB RAM,...      OnePlus   
2  OnePlus Bullets Z2 Bluetooth Wireless in Ear E...      OnePlus   
3  Samsung Galaxy M33 5G (Mystique Green, 6GB, 12...      Samsung   
4  OnePlus Nord CE 2 Lite 5G (Black Dusk, 6GB RAM...      OnePlus   

         main_category     sub_category  \
0  tv, audio & cameras  All Electronics   
1  tv, audio & cameras  All Electronics   
2  tv, audio & cameras  All Electronics   
3  tv, audio & cameras  All Electronics   
4  tv, audio & cameras  All Electronics   

                                               image  \
0  https://m.media-amazon.com/images/I/81eM15lVcJ...   
1  https://m.media-amazon.com/images/I/71AvQd3Vzq...   
2  https://m.media-amazon.com/images/I/51UhwaQXCp...   
3  https://m.media-amazon.com/images/I/81I3w4J6yj...   
4  https://m.media-amazon.com/images/I/71V--WZVUI...   

                                                link  ratings  no_of_ratings  \
0  https://www.amazon.in/Redmi-Power-Black-128GB-...      4.0          965.0   
1  https://www.amazon.in/OnePlus-Nord-Lite-128GB-...      4.3       113956.0   
2  https://www.amazon.in/Oneplus-Bluetooth-Wirele...      4.2        90304.0   
3  https://www.amazon.in/Samsung-Mystique-Storage...      4.1        24863.0   
4  https://www.amazon.in/OnePlus-Nord-Black-128GB...      4.3       113956.0   

   discount_price  actual_price  correct_no_of_ratings  
0         10999.0       18999.0                   True  
1         18999.0       19999.0                   True  
2          1999.0        2299.0                   True  
3         15999.0       24999.0                   True  
4         18999.0       19999.0                   True

In [21]:
# sorting the manufactures according to their prices
DS[['actual_price','manufacturer']].groupby('manufacturer').mean().round(2).sort_values(by='actual_price',ascending=False)

actual_price
manufacturer               
Havells-Lloyd      99990.00
Vu                 80000.00
Hisense            70990.00
VU                 61250.00
Apple              60810.18
...                     ...
Mungyo               182.00
Maped                165.00
ShopsGoods           120.00
Apsara                99.86
Faber                 50.00

[1758 rows x 1 columns]

In [ ]:
#add column with discount and discount_percentage
DS['discount']=DS['actual_price']-DS['discount_price']
DS['discount_percentage']=round((1-DS['discount']/DS['actual_price']*100),2)
DS.head()

name manufacturer  \
0  Redmi 10 Power (Power Black, 8GB RAM, 128GB St...        Redmi   
1  OnePlus Nord CE 2 Lite 5G (Blue Tide, 6GB RAM,...      OnePlus   
2  OnePlus Bullets Z2 Bluetooth Wireless in Ear E...      OnePlus   
3  Samsung Galaxy M33 5G (Mystique Green, 6GB, 12...      Samsung   
4  OnePlus Nord CE 2 Lite 5G (Black Dusk, 6GB RAM...      OnePlus   

         main_category     sub_category  \
0  tv, audio & cameras  All Electronics   
1  tv, audio & cameras  All Electronics   
2  tv, audio & cameras  All Electronics   
3  tv, audio & cameras  All Electronics   
4  tv, audio & cameras  All Electronics   

                                               image  \
0  https://m.media-amazon.com/images/I/81eM15lVcJ...   
1  https://m.media-amazon.com/images/I/71AvQd3Vzq...   
2  https://m.media-amazon.com/images/I/51UhwaQXCp...   
3  https://m.media-amazon.com/images/I/81I3w4J6yj...   
4  https://m.media-amazon.com/images/I/71V--WZVUI...   

                                                link  ratings  no_of_ratings  \
0  https://www.amazon.in/Redmi-Power-Black-128GB-...      4.0          965.0   
1  https://www.amazon.in/OnePlus-Nord-Lite-128GB-...      4.3       113956.0   
2  https://www.amazon.in/Oneplus-Bluetooth-Wirele...      4.2        90304.0   
3  https://www.amazon.in/Samsung-Mystique-Storage...      4.1        24863.0   
4  https://www.amazon.in/OnePlus-Nord-Black-128GB...      4.3       113956.0   

   discount_price  actual_price  correct_no_of_ratings  discount  \
0         10999.0       18999.0                   True    8000.0   
1         18999.0       19999.0                   True    1000.0   
2          1999.0        2299.0                   True     300.0   
3         15999.0       24999.0                   True    9000.0   
4         18999.0       19999.0                   True    1000.0   

   discount_percentage  
0                 0.58  
1                 0.95  
2                 0.87  
3                 0.64  
4                 0.95

**Popular consumer electronics brand**

In [23]:
values=DS['manufacturer'].value_counts().keys().tolist()[:15]
counts=DS['manufacturer'].value_counts().tolist()[:15]
counts,values

plt = px.bar(DS, x=values, y=counts, color_discrete_sequence=['BLUE']*len(DS))

plt.update_layout(
        plot_bgcolor='WHITE',
        xaxis_title='Name of The Brand',
        yaxis_title='Count',
        title='<b>Popular Electronics Brands</b>'
)

plt.show()


**As We show above, 'TheGiftKart' is the most popular Electronics brand .**

In [24]:
#check minimum discount price row
DS[DS['discount']==DS['discount'].min()]

name manufacturer  \
4431  HP Laserjet Pro M126nw Multi-Function Monochro...           HP   

            main_category     sub_category  \
4431  tv, audio & cameras  All Electronics   

                                                  image  \
4431  https://m.media-amazon.com/images/I/61d9Rb8EDZ...   

                                                   link  ratings  \
4431  https://www.amazon.in/HP-LaserJet-M126nw-Multi...      4.2   

      no_of_ratings  discount_price  actual_price  correct_no_of_ratings  \
4431         5256.0         21328.0      21328.88                   True   

      discount  discount_percentage  
4431      0.88                  1.0

In [25]:
#check maximum actual_price
DS[DS['actual_price']==DS['actual_price'].max()]

name manufacturer  \
6027  Samsung Galaxy S23 Ultra 5G (Green, 12GB, 512G...      Samsung   

            main_category     sub_category  \
6027  tv, audio & cameras  All Electronics   

                                                  image  \
6027  https://m.media-amazon.com/images/I/61VfL-aiTo...   

                                                   link  ratings  \
6027  https://www.amazon.in/Samsung-Galaxy-Ultra-Gre...      4.4   

      no_of_ratings  discount_price  actual_price  correct_no_of_ratings  \
6027           91.0        134999.0      161999.0                   True   

      discount  discount_percentage  
6027   27000.0                 0.83

In [26]:
#sort based on rating
manufacturing_rating=DS.groupby('manufacturer')['ratings'].mean()
top_manufacturer=manufacturing_rating.sort_values(ascending=False).head(50)
print(top_manufacturer)

manufacturer
XGMO®360             5.000000
MKY                  5.000000
HARRY                5.000000
Yorten               5.000000
USB-C                5.000000
TYCON                5.000000
A3sprime             5.000000
Marcelle             5.000000
Pecan™               5.000000
Magnetic             5.000000
ORUM                 5.000000
Uvasaggaharam        5.000000
TOPRONICS            5.000000
MOVCTON              5.000000
Noise_Replacement    5.000000
18                   5.000000
KXXO®                5.000000
LTZ                  5.000000
LKMO                 5.000000
Squalo               5.000000
Virtutron            5.000000
B                    5.000000
KimTok               5.000000
Artikel              4.900000
Vichaxan             4.900000
JIYAMART             4.800000
inari                4.800000
BARROT               4.800000
Mavro                4.750000
Rylan                4.700000
DIZORO               4.700000
ProTVSolutions       4.700000
AFPIN                4.7000

In [27]:
# new DATAFRAME with only manufacturer and ratings columns
sel_cols=['manufacturer','ratings']
new_ds=DS[sel_cols]
new_ds


manufacturer  ratings
0           Redmi      4.0
1         OnePlus      4.3
2         OnePlus      4.2
3         Samsung      4.1
4         OnePlus      4.3
...           ...      ...
9595       PALAY®      4.1
9596         MOCA      4.2
9597    ZEBRONICS      3.5
9598       LINQS®      3.6
9599      Pragati      4.5

[9002 rows x 2 columns]

In [28]:
# sorting manufacturers with avrage ratings
new_ds1=new_ds.groupby('manufacturer')
new_ds1.mean()



ratings
manufacturer          
(Renewed)     3.597222
1             4.100000
10W           3.700000
10WeRun       4.050000
18            5.000000
...                ...
worldmacs     4.200000
{2022         3.100000
‎GIZGA        2.500000
【2            4.700000
𝟐𝟎𝟐𝟐          3.200000

[1758 rows x 1 columns]

In [29]:
# filtering manufacturer name
manufacturer_name=values
filtered_ds=new_ds[new_ds['manufacturer'].isin(manufacturer_name)]

print(filtered_ds)
filtered_ds['manufacturer'].unique()


      manufacturer  ratings
3          Samsung      4.1
6             boAt      3.9
8       Fire-Boltt      4.3
9          Samsung      4.1
13            boAt      3.8
...            ...      ...
9568            HP      4.2
9576  AmazonBasics      4.1
9578   TheGiftKart      4.2
9591        Spigen      4.6
9597     ZEBRONICS      3.5

[2135 rows x 2 columns]


array(['Samsung', 'boAt', 'Fire-Boltt', 'SanDisk', 'Portronics',
       'ZEBRONICS', 'Noise', 'HP', 'Sounce', 'AmazonBasics', 'Amazon',
       'Spigen', 'TheGiftKart', 'Pikkme', 'CEDO'], dtype=object)

In [30]:
#check mean of top manufacturer ratings
new_ds1=filtered_ds.groupby('manufacturer')
new_ds1.mean()

ratings
manufacturer          
Amazon        4.026087
AmazonBasics  4.156034
CEDO          4.085542
Fire-Boltt    4.166667
HP            4.016000
Noise         3.904444
Pikkme        4.101235
Portronics    4.067188
Samsung       4.164539
SanDisk       4.337864
Sounce        4.028571
Spigen        4.309852
TheGiftKart   4.323985
ZEBRONICS     3.883212
boAt          4.026360

# Conclusion:
Based on the analysis of Electronics sales dataset, some of the key observations shown below:
Your analysis of the Electronics sales dataset highlights several key insights:

1. **Customer Ratings Reliability:** An impressive 98.77% of customers have given an average rating of 4.0, indicating the authenticity of the received ratings. This suggests genuine customer feedback, devoid of falsification.

2. **Top Manufacturers' Ratings:** Among the top 15 manufacturers, products demonstrate favorable average ratings, hovering around 4.2. This trend underscores a positive customer sentiment towards these manufacturers, indicating consistently satisfying experiences.

3. **Popularity vs. Ratings:** High ratings for popular products signify a customer inclination towards well-established brands. However, a crucial revelation emerges: while popular products often receive high ratings, lesser-known manufacturers also produce highly rated products. This indicates that quality isn't solely tied to brand popularity.

In summary, the analysis reaffirms the reliability of customer ratings for Electronics products. It emphasizes the prevalence of positive ratings among popular items but underscores the presence of highly rated products from both renowned and less familiar brands. This insight encourages a holistic consideration of various manufacturers, ensuring access to quality products across diverse brand landscapes.